be/src/exec/pipeline/query_context.cpp

```diff
@@ -28,43 +28,116 @@ void QueryContext::cancel(const Status& status) {
     _fragment_mgr->cancel(status);
 }
 
-QueryContextManager::QueryContextManager() = default;
+static constexpr size_t QUERY_CONTEXT_MAP_SLOT_NUM = 64;
+QueryContextManager::QueryContextManager()
+        : _mutexes(QUERY_CONTEXT_MAP_SLOT_NUM),
+          _context_maps(QUERY_CONTEXT_MAP_SLOT_NUM),
```

Comment:
Shard map? Why not a concurrent map?

Reply:
Otherwise the shard map could be extracted to a common class?

Reply:
very good suggest, but concurrent map in this PR,  a mutex manage to two map, it's not a general-purpose concurrent map.